java/ci/build-in-docker.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+set -e
+gcc --version
+
+SIGN_FILE=$1
+#Set absolute path for OUT_PATH
+OUT_PATH=$WORKSPACE/$OUT
+
+# set on Jenkins parameter
+if [ -z $RMM_VERSION ]
+then
+RMM_VERSION=`git describe --tags | grep -o -E '([0-9]+\.[0-9]+)'`
+fi
+echo "RMM_VERSION: $RMM_VERSION, SIGN_FILE: $SIGN_FILE, SKIP_JAVA_TESTS: $SKIP_JAVA_TESTS,\
+ BUILD_CPP_TESTS: $BUILD_CPP_TESTS, ENABLED_PTDS: $ENABLE_PTDS, OUT_PATH: $OUT_PATH"
+
+INSTALL_PREFIX=/usr/local/rapids
+export GIT_COMMITTER_NAME="ci"
+export GIT_COMMITTER_EMAIL="ci@nvidia.com"
+export CUDACXX=/usr/local/cuda/bin/nvcc
+export RMM_ROOT=$INSTALL_PREFIX
+export DLPACK_ROOT=$INSTALL_PREFIX
+export LIBCUDF_KERNEL_CACHE_PATH=/rapids
+
+cd /rapids/
+git clone --recurse-submodules https://github.com/rapidsai/rmm.git -b branch-$RMM_VERSION
+git clone --recurse-submodules https://github.com/rapidsai/dlpack.git -b cudf
+
+###### Build rmm/dlpack ######
+mkdir -p /rapids/rmm/build
+cd /rapids/rmm/build
+echo "RMM SHA: `git rev-parse HEAD`"
+cmake .. -DCMAKE_CXX11_ABI=OFF -DCMAKE_INSTALL_PREFIX=$INSTALL_PREFIX -DBUILD_TESTS=$BUILD_CPP_TESTS
```

Comment:
Per an offline discussion with @kkraus14 we shouldn't need to explicitly off the CXX11 ABI, as devtoolset is already building binaries for an OS that's older than that ABI change.

This cmake option is being actively removed from the projects, so it should be removed from this cmake command along with the ones down below.

Reply:
Working on removing it in #6209

Reply:
Wait for #6209. And need to remove abiOff profile in pom.

Reply:
I filed #6223 to remove the CXX11 ABI flag from the pom and JNI cmake.